Fix reduction over dim list for empty input

manuelcandales opened the pull request:
Summary:
Allow empty input in `MapReduceOverDimListPlan`
Avoid integer division by zero in `parallel_for_each_reduce_over_dim_list_output_index`
Add empty input tests for ops: any, mean, sum & var

Differential Revision: D81383049

swolchok reviewed (changes requested):
Review automatically exported from Phabricator review in Meta.

manuelcandales committed:
Fix reduction over dim list for empty input (#13833)

Summary:

Allow empty input in `MapReduceOverDimListPlan`
Avoid integer division by zero in `parallel_for_each_reduce_over_dim_list_output_index`
Add empty input tests for ops: any, mean, sum & var

Differential Revision: D81383049
